packages/wp-now/src/tests/wp-now.spec.ts

```diff
@@ -144,3 +151,137 @@ test('isWordPressDevelopDirectory returns false for incomplete WordPress-develop
+	let tmpExampleDirectory;
+
+	/**
+	 * Copy example directory to a temporary directory
+	 *
+	 * Also, silent the console.log() output.
+	 */
+	beforeEach(() => {
+		const tmpDirectory = os.tmpdir();
+		const directoryHash = crypto.randomBytes(20).toString('hex');
+
+		tmpExampleDirectory = path.join(tmpDirectory, directoryHash);
+		fs.ensureDirSync(tmpExampleDirectory);
+		fs.copySync(exampleDir, tmpExampleDirectory);
+
+		jest.spyOn(console, 'log').mockImplementation(() => {});
+	});
+
+	/**
+	 * Remove temporary directory
+	 */
+	afterEach(() => {
+		fs.rmSync(tmpExampleDirectory, { recursive: true, force: true });
+	});
+
+	/**
+	 * Expect that all provided mount point paths are empty directories which are result of file system mounts.
+	 *
+	 * @param mountPaths List of mount point paths that should exist on file system.
+	 * @param projectPath Project path.
+	 */
+	const expectEmptyMountPoints = (mountPaths, projectPath) => {
+		mountPaths.map((relativePath) => {
+			const fullPath = path.join(projectPath, relativePath);
+
+			expect(fs.existsSync(fullPath)).toBe(true);
+			expect(fs.readdirSync(fullPath)).toEqual([]);
+			expect(fs.lstatSync(fullPath).isDirectory()).toBe(true);
+		});
+	};
+
+	/**
+	 * Expect that all required files exist for PHP.
+	 *
+	 * @param requiredFiles List of files that should be accessible by PHP.
+	 * @param documentRoot Document root of the PHP server.
+	 * @param php NodePHP instance.
+	 */
+	const expectRequiredFiles = (requiredFiles, documentRoot, php) => {
+		requiredFiles.map((relativePath) => {
+			const fullPath = path.join(documentRoot, relativePath);
+			expect(php.fileExists(fullPath)).toBe(true);
+		});
+	};
+
+	/**
+	 * Test that startWPNow in "index" mode doesn't change anything in the project directory.
+	 */
+	test('startWPNow starts index mode', async () => {
+		const projectPath = exampleDir + '/index';
+		const initialProjectContent = fs.readdirSync(projectPath);
+
+		const rawOptions: Partial<WPNowOptions> = {
+			projectPath: projectPath,
+		};
+
+		await startWPNow(rawOptions);
+
+		const finalProjectContent = fs.readdirSync(projectPath);
+		expect(initialProjectContent).toEqual(finalProjectContent);
+	});
+
+	/**
+	 * Test that startWPNow in "wp-content" mode mounts required files and directories, and
+	 * that required files exist for PHP.
+	 */
+	test('startWPNow starts wp-content mode', async () => {
+		const projectPath = path.join(tmpExampleDirectory, 'wp-content');
+
+		const rawOptions: Partial<WPNowOptions> = {
+			projectPath: projectPath,
+		};
+
+		const { php, options: wpNowOptions } = await startWPNow(rawOptions);
+
+		const mountPointPaths = [
+			'database',
+			'db.php',
+			'mu-plugins',
+			'plugins/sqlite-database-integration',
+		];
+
+		expectEmptyMountPoints(mountPointPaths, projectPath);
+
+		const requiredFiles = [
+			'wp-content/db.php',
+			'wp-content/mu-plugins/0-allow-wp-org.php'
+		];
+
+		expectRequiredFiles(requiredFiles, wpNowOptions.documentRoot, php);
```

Comment:
Are there any files we don't expect to be placed in the project directory? Can we add tests for them?

Reply:
The `expectEmptyMountPoints` function will ensure that mount points are empty directories, so those files can't exist if that part passes. Do you have any other file in mind?

Reply:
Oh, as an example, that `wp-config.php` doesn't accidentally exist in the project directory. It'd probably be good to have assertions for each file that `wp-now` might touch.